www/js/components/point-map.js

```diff
@@ -59,21 +56,26 @@ class PointMap extends Component {
     dispatch(setMapCenter(this.state.center));
   }
 
-  onMapMove(leaflet) {
-    const { dispatch } = this.props;
-    const { lat, lng } = leaflet.target.getCenter();
-    this.setState({center:[lat, lng]});
-  }
-
   onMapMoved(leaflet) {
-    const { dispatch } = this.props;
     const { lat, lng } = leaflet.target.getCenter();
-    this.setState({zoom:leaflet.target.getZoom(), center:[lat, lng]});
-    dispatch(setMapCenter(this.state.center));
+    const { _northEast, _southWest } = leaflet.target.getBounds();
+    const diffLat = Math.abs(_northEast.lat - _southWest.lat);
+    const diffLng = Math.abs(_northEast.lat - _southWest.lng);
+    this.setState({
+      zoom:leaflet.target.getZoom(),
+      center:[lat, lng],
+      box:[
+        _northEast.lat + diffLat,
+        _northEast.lng + diffLng,
+        _southWest.lat - diffLat,
+        _southWest.lng - diffLng
+      ]});
```

Comment:
You should test passing `this.props.afterMoved.bind(this, leaflet)` as the second argument to `setState()`.
This has the benefit of running the afterMoved callback after react has rendered `<PointMap />` with the state above.